File: python/colas/colaArr.py
MAXSIZE = 5
queue = [0] * MAXSIZE
front = -1
rear = -1

def insertar():
    global front, rear
    
    elemento = int(input("\nIngrese el elemento: "))
    
    if rear == MAXSIZE - 1:
        print("\nDESBORDAMIENTO (OVERFLOW)\n")
        return
    
    if front == -1 and rear == -1:
        front = rear = 0
    else:
        rear += 1
    
    queue[rear] = elemento
    print("\nElemento insertado correctamente.\n")

def eliminar():
    global front, rear
    
    if front == -1 or front > rear:
        print("\nSUBDESBORDAMIENTO (UNDERFLOW)\n")
        return
    
    elemento = queue[front]
    
    if front == rear:
        front = rear = -1
    else:
        front += 1
    
    print(f"\nElemento eliminado: {elemento}\n")

def mostrar():
    if rear == -1 or front == -1 or front > rear:
        print("\nLa cola está vacía.\n")
    else:
        print("\nElementos en la cola:\n")
        for i in range(front, rear + 1):
            print(f"{queue[i]}\n")

def main():
    opcion = 0
    
    while opcion != 4:
        print("\n****************** MENÚ PRINCIPAL ******************\n")
        print("====================================================\n")
        print("1. Insertar un elemento\n")
        print("2. Eliminar un elemento\n")
        print("3. Mostrar la cola\n")
        print("4. Salir\n")
        
        opcion = int(input("Ingrese su opción: "))
        
        if opcion == 1:
            insertar()
        elif opcion == 2:
            eliminar()
        elif opcion == 3:
            mostrar()
        elif opcion == 4:
            print("\nSaliendo del programa...\n")
        else:
            print("\nOpción inválida. Intente nuevamente.\n")

if __name__ == "__main__":
    main()

File: jv/colas/colaArreglo/ColaArr.java
import java.util.Scanner;

public class ColaArr {
    private static final int MAXSIZE = 5;
    private static int[] queue = new int[MAXSIZE];
    private static int front = -1, rear = -1;
    
    public static void insertar() {
        Scanner sc = new Scanner(System.in);
        int elemento;
        System.out.print("\nIngrese el elemento: ");
        elemento = sc.nextInt();
        
        if (rear == MAXSIZE - 1) {
            System.out.println("\nDESBORDAMIENTO (OVERFLOW)\n");
            return;
        }
        
        if (front == -1 && rear == -1) {
            front = rear = 0;
        } else {
            rear++;
        }
        
        queue[rear] = elemento;
        System.out.println("\nElemento insertado correctamente.\n");
    }
    
    public static void eliminar() {
        if (front == -1 || front > rear) {
            System.out.println("\nSUBDESBORDAMIENTO (UNDERFLOW)\n");
            return;
        }
        
        int elemento = queue[front];
        
        if (front == rear) {
            front = rear = -1;
        } else {
            front++;
        }
        
        System.out.println("\nElemento eliminado: " + elemento + "\n");
    }
    
    public static void mostrar() {
        if (rear == -1 || front == -1 || front > rear) {
            System.out.println("\nLa cola está vacía.\n");
        } else {
            System.out.println("\nElementos en la cola:\n");
            for (int i = front; i <= rear; i++) {
                System.out.println(queue[i] + "\n");
            }
        }
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int opcion = 0;
        
        while (opcion != 4) {
            System.out.println("\n****************** MENÚ PRINCIPAL ******************\n");
            System.out.println("====================================================\n");
            System.out.println("1. Insertar un elemento\n");
            System.out.println("2. Eliminar un elemento\n");
            System.out.println("3. Mostrar la cola\n");
            System.out.println("4. Salir\n");
            System.out.print("Ingrese su opción: ");
            opcion = sc.nextInt();
            
            switch (opcion) {
                case 1:
                    insertar();
                    break;
                case 2:
                    eliminar();
                    break;
                case 3:
                    mostrar();
                    break;
                case 4:
                    System.out.println("\nSaliendo del programa...\n");
                    break;
                default:
                    System.out.println("\nOpción inválida. Intente nuevamente.\n");
            }
        }
        
        sc.close();
    }
}